src/Altinn.App.Core/Internal/Process/ProcessEventDispatcher.cs

```diff
@@ -13,39 +10,33 @@ namespace Altinn.App.Core.Internal.Process;
 /// <summary>
 /// Default implementation of the process event dispatcher
 /// </summary>
-class ProcessEventDispatcher : IProcessEventDispatcher
+public class ProcessEventDispatcher : IProcessEventDispatcher
 {
     private readonly IInstanceClient _instanceClient;
     private readonly IInstanceEventClient _instanceEventClient;
-    private readonly ITaskEvents _taskEvents;
-    private readonly IAppEvents _appEvents;
     private readonly IEventsClient _eventsClient;
-    private readonly bool _registerWithEventSystem;
+    private readonly IOptions<AppSettings> _appSettings;
     private readonly ILogger<ProcessEventDispatcher> _logger;
 
-    public ProcessEventDispatcher(
-        IInstanceClient instanceClient, 
+    /// <summary>
+    /// Default implementation of the process event dispatcher
+    /// </summary>
+    public ProcessEventDispatcher(IInstanceClient instanceClient,
```

Comment:
Refactored ProcessEventDispatcher to only dispatch events. Handling of events was moved to ProcessEventHandlingDelegator, which forwards the events to the correct event handler.